### PR TITLE
Repaired category detail sort

### LINKED ISSUE #1307: Possibility to sort the motions within a category is broken
`…/motions/categories/…/sort`

**Problem:**
If you click in the detail view of a category in the three-point menu on `Sort motions` , the following page appears and it is not possible to sort the motions.

**Solution:**
The Sort Motions page appears correctly.

Bildschirmfoto 2022-06-22 um 18 51 15

### PR DESCRIPTION
**Pull request by luisa-beerboom**
Closes #1307

### CODE REVIEW

File: client/src/app/site/pages/meetings/pages/motions/pages/categories/components/category-detail-sort/category-detail-sort.component.ts [hunk 1]
```diff
@@ -88,34 +95,14 @@ export class CategoryDetailSortComponent extends BaseMeetingComponent implements
...
-                this.motionsCount = filtered.length;
-                if (this.motionsCopy.length === 0) {
-                    this.initializeList(filtered);
-                } else {
-                    this.motionsSubject.next(this.handleMotionUpdates(filtered));
-                }
-            })
-        );
+        //this.createMotionSubscriptions({onInit: true});
+        console.log(`viewInit`);
```
Comment (jsangmeister) [inline client/src/app/site/pages/meetings/pages/motions/pages/categories/components/category-detail-sort/category-detail-sort.component.ts:99]: remove

File: client/src/app/site/pages/meetings/pages/motions/pages/categories/components/category-detail-sort/category-detail-sort.component.ts [hunk 2]
```diff
@@ -88,34 +95,14 @@ export class CategoryDetailSortComponent extends BaseMeetingComponent implements
...
-     *
-     * @param motions An array containing the new motions.
-     */
-    private initializeList(motions: ViewMotion[]): void {
-        motions.sort((a, b) => a.category_weight - b.category_weight);
-        this.motionsSubject.next(motions);
-        this.motionsCopy = motions;
+    public onIdFound(id: Id | null): void {
+        this._categoryId = id;
+
```
Comment (jsangmeister) [inline client/src/app/site/pages/meetings/pages/motions/pages/categories/components/category-detail-sort/category-detail-sort.component.ts:104]: remove

**Review by jsangmeister — CHANGES_REQUESTED** [review]
looks good besides some debug stuff. Not tested

File: client/src/app/site/pages/meetings/pages/motions/pages/categories/components/category-detail-sort/category-detail-sort.component.ts (hunk 2 above)
Comment (luisa-beerboom) [reply client/src/app/site/pages/meetings/pages/motions/pages/categories/components/category-detail-sort/category-detail-sort.component.ts:104]: Remove what?

**Review by emanuelschuetze — CHANGES_REQUESTED** [review]
Create two categories, move one category as subcategory. Add motions to subcategory. Open subcategory detail view => empty, no motions!

**Review by emanuelschuetze — APPROVED** [review]

**Pull request merged**